app/static/js/admin/media/upload.js

```diff
@@ -23,8 +23,47 @@ $(function() {
 
     /* let the `btn-upload` div act as the submit button */
     $(".btn-upload").click(function () {
-        $('form.upload-form').submit();
-        return false;
+        var formData = new FormData($('.upload-form')[0]);
+        $.ajax({
+            url: '/admin/media/upload',  //server script to process data
+            type: 'POST',
+            xhr: function() {  // custom xhr
+                myXhr = $.ajaxSettings.xhr();
+                return myXhr;
+            },
+            //Ajax events
+            success: function(data, textStatus, jqXHR) {
+                var response = jQuery.parseJSON(jqXHR.responseText);
+                if (response.extension == null && response.filename == null){
```

Comment:
Can you explain what's happening here?  If we're in the success function, when would the extension / filename be null?